src/dune_rules/preprocessing.ml

```diff
@@ -1,101 +1,6 @@
 open Import
 open Memo.O
-
-(* Encoded representation of a set of library names + scope *)
-module Key : sig
```

Comment:
How come you needed to split this into a separate module?

Reply:
`Action_unexpanded` would need `Preprocessing` now. And `Preprocessing` already depends on `Action_unexpanded`.